### PR TITLE
Transpile Loops

### PR DESCRIPTION
**Pull request by typ3c4t**

### CODE REVIEW

File: transpiler/src/zero_generator.rs [hunk 1]
```diff
@@ -136,33 +142,55 @@ impl Transpiler {
         self.add_instruction(GeneralInstruction::Pseudo(jump_inst));
     }
 
+    fn add_jump_if(&mut self, cond: LocalOrImmediate, addr: ImmediateOrMacro) {
```
Comment (npwardberkeley) [inline transpiler/src/zero_generator.rs:145]: Hmm I was imagining that mostly we'd just construct instructions and then call the more generic `add_instruction`; but this way works too! There's pros and cons

File: transpiler/src/zero_generator.rs [hunk 2]
```diff
@@ -136,33 +142,55 @@ impl Transpiler {
...
+        let post = format!("post{}", self.label_count);
+        self.label_count += 1;
+        (pre, post)
     }
 
+    fn new_variable(&mut self) -> String {
+        let name = format!("var{}", self.variable_count);
+        self.variable_count += 1;
+        name
+    }
```
Comment (npwardberkeley) [inline transpiler/src/zero_generator.rs:177]: nit: newline

File: transpiler/src/zero_generator.rs [hunk 3]
```diff
@@ -136,33 +142,55 @@ impl Transpiler {
         self.add_instruction(GeneralInstruction::Pseudo(jump_inst));
     }
 
+    fn add_jump_if(&mut self, cond: LocalOrImmediate, addr: ImmediateOrMacro) {
+        let jump_inst = PseudoInstruction::JumpIf { cond, addr };
+        self.add_instruction(GeneralInstruction::Pseudo(jump_inst));
+    }
+
+    fn add_initialize(&mut self, name: String) {
+        let init = PseudoInstruction::Init { name };
```
Comment (npwardberkeley) [inline transpiler/src/zero_generator.rs:151]: What exactly is an initialize?

File: transpiler/src/zero_generator.rs [hunk 4]
```diff
@@ -247,6 +294,28 @@ impl Transpiler {
         }
     }
 
+    fn transpile_repeat(&mut self, op: &ExprRepeat) {
+
```
Comment (npwardberkeley) [inline transpiler/src/zero_generator.rs:298]: nit: extra whitespace

File: transpiler/src/zero_generator.rs (hunk 1 above)
Comment (typ3c4t) [reply transpiler/src/zero_generator.rs:145]: sounds good; we can always shift back if need be

**Pull request merged**